sdk/typescript/src/utils/iota-types.ts

```diff
@@ -80,22 +80,39 @@ export function normalizeStructTag(type: string | StructTag): string {
 }
 
 /**
+ * Normalize an IOTA address to ensure consistent format.
  * Perform the following operations:
  * 1. Make the address lower case
  * 2. Prepend `0x` if the string does not start with `0x`.
  * 3. Add more zeros if the length of the address(excluding `0x`) is less than `IOTA_ADDRESS_LENGTH`
- *
+ * 4. Validate that the address contains only hex characters
```

Comment:
I would not make the validation as a step of the normalization, is something extra we can explain below